### PR TITLE
Fixing the sync-incremental-overwrite connection options

### PR DESCRIPTION
**Pull request by himanshudube97**

### CODE REVIEW

File: src/components/Connections/CreateConnectionForm.tsx [hunk 1]
```diff
@@ -459,6 +459,7 @@ const CreateConnectionForm = ({
                         <Switch
                           data-testid={`incremental-all-streams`}
                           checked={incrementalAllStreams}
+                          disabled={sourceStreams[0]?.cursorField ? false: true}
```
Comment (fatchat) [inline src/components/Connections/CreateConnectionForm.tsx:462]: not understanding this check... why is `sourceStreams[0]` special?

File: src/components/Connections/CreateConnectionForm.tsx [hunk 2]
```diff
@@ -490,6 +491,7 @@ const CreateConnectionForm = ({
                           <Switch
                             data-testid={`stream-sync-${idx}`}
                             checked={stream.selected}
+                            disabled={stream.syncMode === "incremental" ? true : false}
```
Comment (fatchat) [inline src/components/Connections/CreateConnectionForm.tsx:494]: no need to enforce this, if they want to deselect a stream they should be able to do so without changing the sync mode first

**Comment by fatchat** [comment]
still need to enforce

3. When the incremental flag is toggled off, select the sync mode to be overwrite. allow the user to change it but this should be the default state on load and on toggle-off



4. When the incremental flag is on, disallow overwrite

File: src/components/Connections/CreateConnectionForm.tsx (hunk 1 above)
Comment (himanshudube97) [reply src/components/Connections/CreateConnectionForm.tsx:462]: -So , the `incremental-**all**-streams` buttons will be disabled if there are **no** cursor fields.

File: src/components/Connections/CreateConnectionForm.tsx (hunk 2 above)
Comment (himanshudube97) [reply src/components/Connections/CreateConnectionForm.tsx:494]: - Got it. I misunderstood overwrite as changing the sync mode (toggle button).
- Overwrite is the option in destination.

**Comment by himanshudube97** [comment]
@fatchat 
- I have added a error toast to tell user that you cant select `incremental` when the sync mode is `overwrite`.

Doubt:
- should i automatically change the sync mode to `append` , if user selects `incremental` and sync mode is already `overwrite`.   (Right now I just show an error, and user change it manually).

- If yes , then should we do it for both `single-incremental` switch and the `all-incremental-switch` both?

File: src/components/Connections/CreateConnectionForm.tsx (hunk 1 above)
Comment (fatchat) [reply src/components/Connections/CreateConnectionForm.tsx]: what i mean is, what if `sourceStreams[0]?.cursorField` is present but `sourceStreams[1]?.cursorField` is not? don't we have to check all the `sourceStreams`

also i don't know if we even want this... should we "disable-all" if even one can't be enabled or if all can't be enabled... i don't know. i just want the logic enforced on a per-stream basis

**Comment by fatchat** [comment]
yes changing to incremental should change to append/dedup

yes do this for "change all streams to incremental" as well

pending: when incremental is turned off, change sync mode to overwrite

and do this for "change all streams to non-incremental" as well

thanks

File: src/components/Connections/CreateConnectionForm.tsx (hunk 2 above)
Comment (himanshudube97) [reply src/components/Connections/CreateConnectionForm.tsx]: -This is removed. User can deselect a stream, when incremental is selected or not.

File: src/components/Connections/CreateConnectionForm.tsx (hunk 1 above)
Comment (himanshudube97) [reply src/components/Connections/CreateConnectionForm.tsx]: - I have added a code to check `isAnyCursorAbsent` .
- Packed inside useMemo with filterSourceStreams as dependency for efficiency.

**Review by fatchat — APPROVED** [review]

**Pull request merged**